code/modules/mob/living/simple_animal/hostile/terror_spiders/procs.dm

```diff
@@ -0,0 +1,447 @@
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/DoShowOrders(var/usecache = 0, var/thequeen = null, var/theempress = null)
+	if (ckey)
+		to_chat(src, "------------------------")
+		if (ai_type == 0)
+			to_chat(src, "Your Orders: <span class='danger'>kill all humanoids on sight! </span>")
+		else if (ai_type == 1)
+			to_chat(src, "Your Orders: <span class='notice'>defend yourself & the hive, without being aggressive </span> ")
+		else if (ai_type == 2)
+			to_chat(src, "Your Orders: <span class='danger'>do not attack anyone, not even in self-defense!</span> ")
+		to_chat(src, "------------------------")
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/BroadcastOrders()
+	var/cache_thequeen = null
+	var/cache_theempress = null
+	for(var/mob/living/simple_animal/hostile/poison/terror_spider/queen/Q in mob_list)
+		if (Q.ckey || !Q.spider_awaymission)
+			cache_thequeen = Q
+			break
+	for(var/mob/living/simple_animal/hostile/poison/terror_spider/empress/E in mob_list)
+		if (E.ckey || !E.spider_awaymission)
+			cache_theempress = E
+			break
+	for(var/mob/living/simple_animal/hostile/poison/terror_spider/S in mob_list)
+		S.DoShowOrders(1,cache_thequeen,cache_theempress)
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/IsInfected(var/mob/B)
```

Comment:
any reason why this needs to be typecasted for all mobs, specifically? Hardly thing that it needs to account for observers.